solutions/src/test/java/ru/sberbank/school/task07/CollectionsTest.java

```diff
@@ -0,0 +1,79 @@
+package ru.sberbank.school.task07;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.Assert;
+
+import java.io.FileNotFoundException;
+import java.lang.reflect.Array;
+import java.util.*;
+
+public class CollectionsTest {
+
+    private String pathToFile;
+    private CollectionsServiceFactoryImpl csfi;
+    private FileParser fileParser;
+
+    @Before
+    public void initialize() {
+        pathToFile = "./src/test/java/ru/sberbank/school/task07/test.txt";
+        csfi = new CollectionsServiceFactoryImpl();
+        fileParser = new FileParserImpl();
+    }
+
+    @Test
+    public void testFileParser() throws FileNotFoundException {
+        initialize();
+        List<String> stringsFile = csfi.getFileParser().parse(pathToFile);
+        Assert.assertNotNull(stringsFile);
+        Assert.assertFalse(stringsFile.isEmpty());
+    }
+
+    @Test
+    public void testCounter() throws FileNotFoundException {
+        int count = csfi.getCounter(fileParser).count(pathToFile);
+        Assert.assertEquals(30, count);
+    }
+
+    @Test
+    public void testDifferentWordsImpl() throws FileNotFoundException {
+        pathToFile = "./src/test/java/ru/sberbank/school/task07/test2.txt";
+        Set set = (Set) csfi.getDifferentWords(fileParser).findSortedDifferentWords(pathToFile);
+        List actual = new ArrayList<>(set);
+        List expected = Arrays.asList("a", "b", "d", "cd", "dc", "qq", "svfdf");
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testWordsFrequency() throws FileNotFoundException {
+        Map actual = (Map) csfi.getWordFrequency(fileParser).countWords(pathToFile);
```

Comment:
Distance between variable &#39;actual&#39; declaration and its first usage is 31, but allowed 3.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1236205644>*